java/src/main/native/src/TableJni.cpp

```diff
@@ -2637,11 +2637,20 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_partition(JNIEnv *env, jc
     auto [partitioned_table, partition_offsets] =
         cudf::partition(*n_input_table, *n_part_column, number_of_partitions);
 
-    // for what ever reason partition returns the length of the result at then
-    // end and hash partition/round robin do not, so skip the last entry for
-    // consistency
     cudf::jni::native_jintArray n_output_offsets(env, output_offsets);
-    std::copy(partition_offsets.begin(), partition_offsets.end() - 1, n_output_offsets.begin());
+    // When row number is 0, the size of partition_offsets will be 0.
+    // Then "partition_offsets.end() - 1" will point to an invalid address,
+    // leading to a crash when copying. So need to check its size before setting the
+    // output_offsets
+    if (partition_offsets.empty()) {
+      // row number is 0. Then fill the output with 0.
+      std::fill(n_output_offsets.begin(), n_output_offsets.end(), 0);
```

Comment:
This basically has the same problem such that I had to reverse my changes in https://github.com/rapidsai/cudf/pull/11709. When the output table is empty, all values in this offsets array will be invalid offset and accessing the output table through these offset values will lead to crash. 

Thus, I'm not sure if we also need to modify `Table.java#getPartitions()` to return an empty array when the table is empty, instead of an this all-zero array? @jlowe What do you think?

Reply:
AFAIK, [the access will not happen](https://github.com/NVIDIA/spark-rapids/blob/branch-22.10/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuPartitioning.scala#L44) because each partition's size is 0 (offset[i+1] - offset[i]), meaning there is no row in a partiton. Do you have verified that empty offset array will lead to crash ?
Returning empty offset array might break the expectation in plugin, but I am not 100% sure.

Reply:
IMO returning an empty offset array is less error-prone than an all-zero array. Of course, empty array can also cause crash due to computing the partition size. Thus, it depends on which way is better to handle for our use case.

Reply:
On the other hand, in the case of Python's dask framework, it doesn't handle zero-size partitions like us thus an all-zero offsets array will lead to a crash thus the offsets array associated with an empty output table also needs to be empty.

Reply:
I just find one place where it requires non-empty offsets array (at least one element). Otherwise, the starting index 1 will be out of bounds.
https://github.com/NVIDIA/spark-rapids/blob/branch-22.10/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuPartitioning.scala#L96

Reply:
We need to do check and test for either way.

Reply:
Alright, since our plugin code can handle zero-size partition, I'm fine to keep this solution (fill out all zero) but please add comments here to clarify.

Reply:
I am fine to return an empty offset array, but we need to double check the breaks in plugin.

Reply:
> When the output table is empty, all values in this offsets array will be invalid offset and accessing the output table through these offset values will lead to crash.

There is no access since the offsets are _not_ invalid.  The partitions are _empty_ (i.e.: partition start == partition end) thus there are no rows to index.  Any code attempting to index without checking there are rows to index is incorrect.  It technically should not matter what value the offsets are, as long as the offsets are all the same value (e.g.: if the offset array is all 7's then it's still a result of all empty tables).

> On the other hand, in the case of Python's dask framework, it doesn't handle zero-size partitions like us

Is that really true?  What if there are 10 rows in the input table and we ask for 20 partitions to be made?  At a minimum 10 of those partitions will be empty.  Even if there are only 10 partitions requested, there's a chance that more than one row will hash to the same partition and thus one of the other partitions will be marked empty (i.e.: partition start == partition end)

> IMO returning an empty offset array is less error-prone than an all-zero array.

I disagree.  The comments clearly state that when requesting N partitions, an offset array of N+1 will be returned, and it's surprising to have a corner case where that is not true.  As I mentioned above, applications already need to handle the possibility of some partitions being empty, and asking to partition an empty table 10 ways should result in 10 empty partitions, not zero partitions.

Reply:
Yeah that sounds reasonable.
So the issue now escalates into a potential dask bug. I'll continue to track it down.